msupply-foundation/open-msupply
# 3319 show actual error if no stock lines are associated with shipment lines to return

Fixes #3319

# 👩🏻‍💻 What does this PR do? 
 

This issue came about from another intermittent issue I appear to be having (#3320) where sometimes the inbound shipment lines don't have an associated stock line, which we need to generate the return.

Anyways, we were expecting the stockLine to always be present, and when it wasn't, we'd get a `please select lines to return` error even when we had selected lines. This PR shows an actual error message :)

![Screenshot 2024-03-21 at 5 48 49 PM](https://github.com/msupply-foundation/open-msupply/assets/55115239/fcedd53f-b8b0-4908-b9b4-986cc9cf495e)


# 🧪 How has/should this change been tested? 


Somehow set up a delivered inbound return with an invoice line with no stock_line_id (maybe just set to NULL in DB?)

- select that line to return
- see error popup

## 💌 Any notes for the reviewer?


In hindsight, I'm wondering if we should instead be sending the inbound shipment line ids to the generate method, and letting that query error if there is no associated stock line? But probably not worth the rework at this point...
